src/Backups/BackupIO_S3.cpp

```diff
@@ -86,8 +86,8 @@ BackupReaderS3::BackupReaderS3(
     const S3::URI & s3_uri_, const String & access_key_id_, const String & secret_access_key_, const ContextPtr & context_)
     : s3_uri(s3_uri_)
     , client(makeS3Client(s3_uri_, access_key_id_, secret_access_key_, context_))
-    , max_single_read_retries(context_->getSettingsRef().s3_max_single_read_retries)
```

Comment:
Backup test failed, but the only relevant change is here. Previously, separate variable was used `max_single_read_retries` instead of `request_settings.max_single_read_retries`. I'll try to reproduce and check if this change matters.

Reply:
Fail is reproducible 100%...